runtime/src/main/java/org/corfudb/runtime/LogReplicationLogicalGroupClient.java

```diff
@@ -0,0 +1,391 @@
+     *
+     * @param logicalGroup The group of tables to be replicated, represented as a string.
+     * @param remoteDestinations The clusters which the specified group will be replicated to.
+     */
+    public void addDestination(String logicalGroup, List<String> remoteDestinations) {
+        Preconditions.checkArgument(isValid(logicalGroup), String.format("[%s] %s", clientName, "logicalGroup is null or empty."));
+        Preconditions.checkArgument(isValid(remoteDestinations), String.format("[%s] %s", clientName, "remoteDestinations is null or empty."));
+        Preconditions.checkArgument(hasNoNullOrEmptyElements(remoteDestinations), String.format("[%s] %s", clientName, "remoteDestinations contains null or empty elements."));
+        List<String> finalRemoteDestinations = new ArrayList<>(deduplicate(remoteDestinations));
+        try {
+            IRetry.build(ExponentialBackoffRetry.class, () -> {
+                try (TxnContext txn = corfuStore.txn(CORFU_SYSTEM_NAMESPACE)) {
+                    ClientDestinationInfoKey clientInfoKey = buildClientDestinationInfoKey(logicalGroup);
+                    DestinationInfoVal clientDestinations = txn.getRecord(sourceMetadataTable, clientInfoKey).getPayload();
+
+                    if (clientDestinations != null) {
+                        finalRemoteDestinations.removeAll(clientDestinations.getDestinationIdsList());
+                        if (finalRemoteDestinations.isEmpty()) {
+                            log.debug("[{}] {}", clientName, "All destinations already present.");
+                            return null;
+                        }
+                        clientDestinations = clientDestinations.toBuilder().addAllDestinationIds(finalRemoteDestinations).build();
+                    } else {
+                        clientDestinations = DestinationInfoVal.newBuilder().addAllDestinationIds(finalRemoteDestinations).build();
+                    }
+
+                    txn.putRecord(sourceMetadataTable, clientInfoKey, clientDestinations, null);
+                    txn.commit();
+                    return null;
+                } catch (TransactionAbortedException tae) {
+                    log.error("[{}] {}: {}", clientName, "Unable to add destinations, will be retried", tae);
+                    throw new RetryNeededException();
+                }
+            }).run();
+        } catch (InterruptedException e) {
+            log.error("[{}] {}: {}", clientName, "Unable to add destinations", e);
+            throw new UnrecoverableCorfuInterruptedError(e);
+        }
+    }
+
+    /**
+     * Remove a destination from the LogReplicationModelMetadataTable for a specific group.
+     *
+     * @param logicalGroup The group of tables that are replicated, represented as a string.
+     * @param destination The cluster from which the logicalGroup should be removed from all future replication.
```

Comment:
I think this would be clearer if reworded as something like - 'cluster which should be removed from the logical group for ...'